### ISSUE TITLE
[Bug]: Wave AI Frequently Timing Out

### ISSUE THREAD

**Issue by xbais** [wavetermdev/waveterm#1108]
### Current Behavior

When I put a search query in the Wave AI window, it simply keeps me waiting for a response and then times out in about 2 minutes. The error shown is this:
![image](https://github.com/user-attachments/assets/2a0106db-6738-4d73-baed-c521e4b1de25)

The same query works on the ChatGPT online website. Interestingly, this has suddenly started happening today... Till yesterday the same was working for me without any issues.

### Expected Behavior

The response should be provided in less than a minute.

### Steps To Reproduce

1. On an Ubuntu 24.04.1 system
2. Download v0.8.12 Appimage for 64 bit system
3. Make the Appimage executable, run it.
4. In the Wave AI window, add a prompt, wait for response

### Wave Version

v0.8.12

### OS

Ubuntu 24.04.1

### Architecture

x64

### Questionnaire

- [X] I'm interested in fixing this myself but don't know where to start
- [ ] I would like to fix and I have a solution
- [ ] I don't have time to fix this right now, but maybe later

**Comment by esimkowitz** [comment]
@xbais are you using the default AI settings? Also how many requests have you been sending? We cap users at 200/day

**Comment by xbais** [comment]
Yes I am using the default settings. And yes I did expect that there will be a cap on number of prompts sent in a day. But what surprised me was that the AI timed out even at the first few prompts for that day. So is it that the prompt counter resets at a specific time of the day that may not match with midnight time of the user's time zone? If so then I have a few requests / suggestions:
1. It would be really great if the AI prompt counter reset at midnight time of the local time zone.
2. Also, the current error message (EC-Timeout) is very confusing, maybe it could be replaced with a more direct message informing the user that they have used up their daily prompt quota.

**Comment by esimkowitz** [comment]
Do you believe you exceeded the 200 requests threshold over a 24 hour period? Unfortunately we can’t do it in the user’s local time zone because we don’t even record this.

**Comment by xbais** [comment]
Yes I did use it a bit the previous day but I dont think I sent 200 requests. 200 is a very large cap for me. I expect that I sent around maximum of 30 requests on the previous day... and about 5 requests the next day after which it started timing out. 

I can try re-creating this issue to check exactly how many requests trigger the time out, if at all it is linked to number of requests...

**Comment by esimkowitz** [comment]
Got it, it's probably something else then. Will investigate further.

**Comment by esimkowitz** [comment]
Could you share your ~/.waveterm/waveapp.log file contents? I want to see if anything looks weird. You can email it to us at support@waveterm.dev

**Comment by xbais** [comment]
> Could you share your ~/.waveterm/waveapp.log file contents? I want to see if anything looks weird. You can email it to us at [support@waveterm.dev](mailto:support@waveterm.dev)

Sent ! Kindly check the email.

**Comment by xbais** [comment]
Interestingly, I didn't use Waveterm for a day and today when I put another prompt in the same chat after the messages that had timed out a few days ago...my new prompts are also timing out still!

**Comment by xbais** [comment]
:star2:  I just closed the old chat window, and opened a fresh Wave AI Chat window in the Waveterm app. I put a prompt in the fresh window, and it worked! :partying_face: :partying_face: 

Hopefully this will provide some clues as to what went wrong...

**Comment by esimkowitz** [comment]
I wonder if your requests were exceeding the token size limit for the model, roughly how complicated were the requests you were asking? Were the inputs or the outputs really large?

**Comment by xbais** [comment]
> I wonder if your requests were exceeding the token size limit for the model, roughly how complicated were the requests you were asking? Were the inputs or the outputs really large?

Okay, so just to sort out the facts...it could be that the expected size of output of the first message was large (I dont remember the prompt now but the prompt irself was just a few lines, and it is highly unlikely that the expected output was too large) but the following should be noted:
1. Once the time-out happened, all further prompts timed out, irrespective of size of the prompts or their expected output size.
2. The last prompt that timed-out was the first prompt that started working when I opened a new Wave AI window.

**Comment by esimkowitz** [comment]
I think we have a lead, will report back

**Comment by esimkowitz** [comment]
This should be fixed now

**Comment by fenris** [comment]
how about if we are using local llm? is it will get hit by token size limit ?

**Comment by fenris** [comment]
2025-03-23 01:22:18.506 [wavesrv] 2025/03/23 01:22:18.506416 sending ai chat message to  endpoint "http://10.0.124.26:11434/v1" using model deepseek-r1:32b
2025-03-23 01:23:13.963 focus win b990cf04-1a94-44de-ac63-2b1c377b91d1
2025-03-23 01:23:13.966 [wavesrv] 2025/03/23 01:23:13.965995 FocusWindow b990cf04-1a94-44de-ac63-2b1c377b91d1
2025-03-23 01:23:13.966 [wavesrv] 2025/03/23 01:23:13.966253 client.WindowIds: [b990cf04-1a94-44de-ac63-2b1c377b91d1]
2025-03-23 01:23:13.969 [service] client.FocusWindow | 6ms
2025-03-23 01:25:55.597 focus win b990cf04-1a94-44de-ac63-2b1c377b91d1
2025-03-23 01:25:55.599 [wavesrv] 2025/03/23 01:25:55.599339 FocusWindow b990cf04-1a94-44de-ac63-2b1c377b91d1
2025-03-23 01:25:55.599 [wavesrv] 2025/03/23 01:25:55.599503 client.WindowIds: [b990cf04-1a94-44de-ac63-2b1c377b91d1]
2025-03-23 01:25:55.601 [service] client.FocusWindow | 4ms
2025-03-23 01:26:04.451 [wavesrv] 2025/03/23 01:26:04.451117 DBGetAllObjsByType table: db_workspace
2025-03-23 01:26:04.451 [wavesrv] 2025/03/23 01:26:04.451214 got workspaces
2025-03-23 01:26:04.451 [wavesrv] 2025/03/23 01:26:04.451219 DBGetAllObjsByType table: db_window
2025-03-23 01:26:41.380 focus win b990cf04-1a94-44de-ac63-2b1c377b91d1
2025-03-23 01:26:41.381 [wavesrv] 2025/03/23 01:26:41.381793 FocusWindow b990cf04-1a94-44de-ac63-2b1c377b91d1
2025-03-23 01:26:41.382 [wavesrv] 2025/03/23 01:26:41.381959 client.WindowIds: [b990cf04-1a94-44de-ac63-2b1c377b91d1]
2025-03-23 01:26:41.395 [service] client.FocusWindow | 15ms
2025-03-23 01:29:53.299 focus win b990cf04-1a94-44de-ac63-2b1c377b91d1
2025-03-23 01:29:53.301 [wavesrv] 2025/03/23 01:29:53.300797 FocusWindow b990cf04-1a94-44de-ac63-2b1c377b91d1
2025-03-23 01:29:53.301 [wavesrv] 2025/03/23 01:29:53.301011 client.WindowIds: [b990cf04-1a94-44de-ac63-2b1c377b91d1]
2025-03-23 01:29:53.302 [service] client.FocusWindow | 3ms
2025-03-23 01:29:57.013 [wavesrv] 2025/03/23 01:29:57.013170 filestore flush: 1/1 entries flushed, err:<nil>
2025-03-23 01:30:02.039 [wavesrv] 2025/03/23 01:30:02.038581 filestore flush: 1/1 entries flushed, err:<nil>
2025-03-23 01:30:07.062 [wavesrv] 2025/03/23 01:30:07.062340 filestore flush: 1/1 entries flushed, err:<nil>
2025-03-23 01:30:12.083 [wavesrv] 2025/03/23 01:30:12.083033 filestore flush: 1/1 entries flushed, err:<nil>
2025-03-23 01:30:17.085 [wavesrv] 2025/03/23 01:30:17.085597 filestore flush: 1/1 entries flushed, err:<nil>
2025-03-23 01:30:21.626 focus win b990cf04-1a94-44de-ac63-2b1c377b91d1
2025-03-23 01:30:21.628 [wavesrv] 2025/03/23 01:30:21.628232 FocusWindow b990cf04-1a94-44de-ac63-2b1c377b91d1
2025-03-23 01:30:21.628 [wavesrv] 2025/03/23 01:30:21.628409 client.WindowIds: [b990cf04-1a94-44de-ac63-2b1c377b91d1]
2025-03-23 01:30:21.641 [service] client.FocusWindow | 15ms
2025-03-23 01:30:27.088 [wavesrv] 2025/03/23 01:30:27.088526 filestore flush: 1/1 entries flushed, err:<nil>
2025-03-23 01:30:32.110 [wavesrv] 2025/03/23 01:30:32.110534 filestore flush: 1/1 entries flushed, err:<nil>
2025-03-23 01:30:37.132 [wavesrv] 2025/03/23 01:30:37.131858 filestore flush: 1/1 entries flushed, err:<nil>
2025-03-23 01:30:42.155 [wavesrv] 2025/03/23 01:30:42.154752 filestore flush: 1/1 entries flushed, err:<nil>
2025-03-23 01:30:52.180 [wavesrv] 2025/03/23 01:30:52.180601 filestore flush: 1/1 entries flushed, err:<nil>
2025-03-23 01:30:57.203 [wavesrv] 2025/03/23 01:30:57.202780 filestore flush: 1/1 entries flushed, err:<nil>
2025-03-23 01:31:09.737 focus win b990cf04-1a94-44de-ac63-2b1c377b91d1
2025-03-23 01:31:09.739 [wavesrv] 2025/03/23 01:31:09.739685 FocusWindow b990cf04-1a94-44de-ac63-2b1c377b91d1
2025-03-23 01:31:09.740 [wavesrv] 2025/03/23 01:31:09.739846 client.WindowIds: [b990cf04-1a94-44de-ac63-2b1c377b91d1]
2025-03-23 01:31:09.742 [service] client.FocusWindow | 5ms
2025-03-23 01:31:17.227 [wavesrv] 2025/03/23 01:31:17.227513 filestore flush: 1/1 entries flushed, err:<nil>
